methods.py

```diff
@@ -156,17 +156,16 @@ def add_module(path):
         module_name = os.path.basename(path)
         module_path = path.replace("\\", "/")  # win32
         modules[module_name] = module_path
-
+ 
     def is_engine(path):
         # Prevent recursively detecting modules in self and other
         # Godot sources when using `custom_modules` build option.
         version_path = os.path.join(path, "version.py")
         if os.path.exists(version_path):
-            with open(version_path) as f:
-                version = {}
-                exec(f.read(), version)
-                if version.get("short_name") == "godot":
-                    return True
+            version = {}
+            execfile(version_path, version)
```

Comment:
Well as you pointed out this doesn't work for Python 3 so it's not a good option :)